src/lib.rs

```diff
@@ -168,6 +168,9 @@ pub trait SubscriberOps {
     /// Removes the subscriber corresponding to `subscriber_id` from the watch list.
     fn remove_subscriber(&mut self, subscriber_id: SubscriberId) -> Result<Self::Subscriber>;
 
+    /// Returns a reference to the subscriber corresponding to `subscriber_id`.
+    fn subscriber_ref(&self, subscriber_id: SubscriberId) -> Option<&Self::Subscriber>;
```

Comment:
Can we return a `Result` here as well so that it in line with the other getter we have for subscribers (i.e. the `subscriber_mut`)?

Reply:
Can you also add a test for when the subscriber was removed, and then we call `subscriber_ref`?